src/gui/src/LogWindow.cpp

```diff
@@ -32,33 +34,58 @@ LogWindow::LogWindow(QWidget *parent) :
     // repeatedly until InputLeap is finished
     setAttribute(Qt::WA_DeleteOnClose, false);
     setupUi(this);
+    
+    // Use a timer to flush the buffer every 100 milliseconds
+    QTimer* timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, this, &LogWindow::flushBuffer);
+    timer->start(100);
```

Comment:
Instad of flushing every 100msec perhaps flush when the buffer size is some size. this will prevent us from doing any extra syncs if the log is still small